bugsnag-android-core/src/main/java/com/bugsnag/android/DeviceData.java

```diff
@@ -61,9 +61,10 @@
 
     @NonNull
     final String[] cpuAbi;
+    private final File dataDirectory;
 
     DeviceData(Connectivity connectivity, Context appContext, Resources resources,
-               String installId, DeviceBuildInfo buildInfo) {
+               String installId, DeviceBuildInfo buildInfo, File dataDirectory) {
```

Comment:
I'm not sure I understand the rationale for this change - it makes it different to the other calculations for memory/orientation, so doesn't seem ideal.

Reply:
This allows for the static method `Environment.getDataDirectory()` to be mocked in tests, which lets us test the value of `device.freeDisk`. If we left it as-is it would also throw a NPE when run in a JVM test as this is an Android-specific test.